### PR TITLE
Add OWNERS_ALIASES to define teams 🥜

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This defines a few teams:
- pipeline-approvers: general OWNERS
- apis-approvers: apis OWNERS
- productivity-approvers: productivity OWNERS (test-infra, scripts, …)
- productivity-reviewers: productivity reviewers 🙃

The idea is to use more the concept of owners *and* reviewers.

<del>This also tries to use OWNERS file to automatically add labels to the
pull-requests that changes a set of files in those folders. This is
enabled by the `owners-labels` prow plugin.</del>

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/cc @afrittoli @bobcatfish @ImJasonH @dlorenc @dibyom @sbwsg for changing some teams if necessary (also, if you think of other areas that could get auto-labels, etc.. it would be nice)

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

**Comment by vdemeester** [comment]
> The following users are mentioned in OWNERS file(s) but are not members of the tektoncd org.
> 
>     * pipelines-approvers
>       
>       * OWNERS

huh… I may need to do that in 2 steps… :man_facepalming:

File: hack/OWNERS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+approvers:
+- productivity-approvers
+reviewrs:
```
Comment (afrittoli) [inline hack/OWNERS:3]: reviewers ?

File: docs/OWNERS [hunk 2]
```diff
@@ -0,0 +1,2 @@
+labels:
+- kind/documentation
```
Comment (afrittoli) [inline docs/OWNERS:2]: uhm, I'm not sure about this.
Any change that includes some docs will be tagged as kind/documentation, so the kind label check will pass and we will forget to set the right label - so in our release notes we'll have a lot of docs changes :D

File: hack/OWNERS (hunk 1 above)
Comment (vdemeester) [reply hack/OWNERS:3]: oups :sweat_smile:

File: docs/OWNERS (hunk 2 above)
Comment (vdemeester) [reply docs/OWNERS:2]: ah, good point, I can remove that for new indeed :upside_down_face:

**Pull request merged**